code/modules/RCD/RCD.dm

```diff
@@ -86,9 +86,9 @@
 		dropped_by.hud_used.toggle_show_schematics_display(null,1, src)
 
 /obj/item/device/rcd/attack_self(var/mob/user)
-	var/turf/T = get_turf(src)
-	if(T?.z != z_last_checked)
-		rebuild_ui()
+	//var/turf/T = get_turf(src)
+	//if(T?.z != z_last_checked) i don't know why this in in here. commented out instead of removed in case it FSU
```

Comment:
The commented-out items are for multi-z purposes: unless they harm the performance of your UI they should probably stay in uncommented.

Reply:
that's not what that's for. looking at the varible it usues it is only used in 2 places, once when it is defined upon item creation, and the second there when it is red. 
it's incredibly jank shitcode that i have no idea the function of and removal seems to affect nothing.
i needed to remove that check to allow redrawing of the UI to display which pipe is currently being selected

Reply:
\>git blame
\>see who wrote it
lmao disregard remove at will